### PR TITLE
[pten] add CI check for using DenseTensor::mutable_data() in pten directions

### PR DESCRIPTION
**Pull request by MingMingShangTian**
### PR types

Others
### PR changes

Others
### Describe

Add CI check for using DenseTensor::mutable_data() in pten directions

### CODE REVIEW

**Review by Shixiaowei02 — DISMISSED** [review]

File: tools/check_file_diff_approvals.sh [hunk 1]
```diff
@@ -262,6 +262,17 @@ if [ "${PTEN_INCLUDE_FLUID_FILES}" != "" ] && [ "${GIT_PR_ID}" != "" ]; then
...
 fi
+PTEN_USE_MUTABLE_DATA_FILES=""
+for CHANGE_FILE in ${HAS_MODIFIED_PTEN_FILES}; do
+    PTEN_DIR_ADDED_LINES=`git diff -U0 upstream/$BRANCH -- ${PADDLE_ROOT}/${CHANGE_FILE} | grep "^+" | grep -w "mutable_data" || true`
+    if [ "${PTEN_DIR_ADDED_LINES}" != "" ] && [ "${GIT_PR_ID}" != "" ]; then
+        PTEN_USE_MUTABLE_DATA_FILES="${PTEN_USE_MUTABLE_DATA_FILES} ${CHANGE_FILE}"
+    fi 
+done
+if [ "${PTEN_USE_MUTABLE_DATA_FILES}" != "" ] && [ "${GIT_PR_ID}" != "" ]; then
+    echo_line="You can not use the DenseTensor::mutable_data() method in paddle/pten files(${PTEN_USE_MUTABLE_DATA_FILES}). If you want to alloc memory, use pten::DeviceContext::Alloc() instead and if you want to get mutable data, use DenseTensor::data(). If you have any questions, you can have one RD (chenwhql, Shixiaowei02, MingMingShangTian, YuanRisheng or zyfncg) review and approva.\n"
```
Comment (chenwhql) [inline tools/check_file_diff_approvals.sh]: approva -> approve

Comment (chenwhql) [reply tools/check_file_diff_approvals.sh]: use pten::DeviceContext::Alloc() or pten::DeviceContext::HostAlloc()

Comment (zyfncg) [inline tools/check_file_diff_approvals.sh]: approval？

Comment (MingMingShangTian) [reply tools/check_file_diff_approvals.sh]: 改为了approve

Comment (MingMingShangTian) [reply tools/check_file_diff_approvals.sh]: Done

**Review by chenwhql — APPROVED** [review]

**Review by Shixiaowei02 — APPROVED** [review]

**Pull request merged**